packages/gatsby-plugin-mdx/src/gatsby-layout-loader.ts

```diff
@@ -59,13 +61,17 @@ const gatsbyLayoutLoader: LoaderDefinition = async function (source) {
   )
 
   if (hasDefaultExport) {
-    return content
+    return res.mdxNode.body
   }
 
   tree.children.unshift({
     type: `mdxjsEsm` as `text`,
     value: `import GatsbyMDXLayout from "${layoutPath}"`,
   })
+  tree.children.push({
+    type: `mdxjsEsm` as `text`,
+    value: `export * from "${layoutPath}"`,
+  })
```

Comment:
This tries to add the pageQuery from the JS layout file to the MDX file. I was hoping that would the pageQuery get executed, cached and the data passed on.

Thats not the case tough :(